storage/src/main/scala/coop/rchain/storage/package.scala

```diff
@@ -59,17 +59,28 @@ package object storage {
     options.sequence[Option, DataCandidate[C, A]]
   }
 
-  /** `consume` does the "consume" thing
+  /** Searches the store for data matching all the given patterns at the given channels.
     *
-    * @param store
-    * @param channels
-    * @param patterns
-    * @param continuation
-    * @param persist
-    * @tparam C a type representing a channel
-    * @tparam P a type representing a pattern
-    * @tparam A a type representing a piece of data
-    * @tparam K a type representing a continuation
+    * If no match is found, then the continuation and patterns are put in the store at the given
+    * channels.
+    *
+    * If a match is found, then the continuation is returned along with the matching data.
+    *
+    * If the matching data has been stored with the `persist` flag set to `true`, then the data
+    * will remain in the store.  Otherwise, the data will be removed from the store.
+    *
+    * If the persist flag is set to `true` and no matching data is found, then the continuation is
+    * put in the store and will remain there after subsequent matches are found.
```

Comment:
Can we shorten this to something like

> If the persist flag is true, the corresponding data or continuation will remain in the store.

Reply:
To answer your more general question and the yr specific one here, this is my attempt to express that things only "stick" if there is no more matching data.  I struggled a bit with how to express this in a reasonably concise manner.

My hope is that the forthcoming tutorials will provide a better elaboration of this idea.

Reply:
Hmm can we make it so that it is more obvious that this might be unexpected behavior? Like

"Warning: Things only "stick" if there is no more matching data. And by the I mean the following:"

and then what you wrote. Because otherwise it seems like you're writing something obvious in an wordy fashion or maybe I just need more sleep.